app/controllers/sessions_controller.rb

```diff
@@ -12,4 +12,9 @@ def create
       render :new
     end
   end
+
+  def destroy
```

Comment:
@iandouglas @AliSchlereth I created the functionality for authentication, but it seems any ideas that I create while logged in as one user, remain for other users or even if I am not logged in. What am I missing in my code? Thanks a lot.

Reply:
@Shniks When you create an idea in your ideas#create, you want to scope that idea to a user (probably the signed in user -- you have a method you built to help with this). Remember an idea belongs to a user.

Reply:
@AliSchlereth Oh okay. In my crazy quest to build the authentication, I almost forgot about that. Let me try and see how that will work. Thanks.

Reply:
@AliSchlereth I guess one way could be to create a foreign key for user_id in the ideas table. Then for the create action in the Ideas Controller, we could do something like:
@user = User.new(params[:user_id]
@idea = Idea.new(idea_params)
@idea.user_id = @user.id
@idea.save

If not above, then do we do it something like @user.ideas.create!.... ? In which case all the tests will need to be re-written. Little confused now.

Reply:
@Shniks Yes, you'll need to update your tests now that you've implemented auth. Rather than use  @user = User.find(params[:user_id)] you want to build it off of the user which is currently signed in. What method do you have (maybe should you have) that helps you know which user is signed in?

When you create an idea you would associate it with a user. So, for example you could call `user.ideas.new(...)` You do not need nested routes for this since you are building it off the signed in user.

Reply:
@Shniks Yep that last message is on the right track. I recommend the @user.ideas.create!....  strategy over the first.

Reply:
Took a look at your schema, right now users aren't connected to anything.  What should users be connected to? What is the relationship (one to many vs many to many)?

Reply:
I know that Users need to be connected to the ideas (as a User has_many ideas but an idea belongs_to a user. I feel I need to have a foreign key for user_id in the Ideas table. Is that right? Should I be creating that key and then working through it? What about the relationship between users (admin) and categories and images? An admin has many images an many categories. So will there be a one to many for those two also? If so, how will I toggle the state between an ordinary user and an admin? Getting confused now. Maybe I need to take baby steps and do the ideas first.